modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayNavigationBar from '@clayui/navigation-bar';
+import {fetch, openToast} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {TFDSView} from './FDSViews';
+import Details from './fds_view/Details';
+
+const NAVIGATION_BAR_ITEMS = [
+	{
+		Component: Details,
+		label: Liferay.Language.get('details'),
+	},
+];
+
+interface IFDSViewProps {
+	fdsViewId: string;
+	fdsViewsAPIURL: string;
+	fdsViewsURL: string;
+	namespace: string;
+}
+
+const FDSView = ({
+	fdsViewId,
+	fdsViewsAPIURL,
+	fdsViewsURL,
+	namespace,
+}: IFDSViewProps) => {
+	const [activeIndex, setActiveIndex] = useState<number>(0);
+	const [fdsView, setFDSView] = useState<TFDSView>();
+	const [loading, setLoading] = useState<boolean>(true);
```

Comment:
Doesn't TS infer types when a default value is set? I think it does, so line 44 and 46 might not need type declaration